CMakeLists.txt

```diff
@@ -13,8 +13,16 @@ enable_testing()
 option(MATERIALX_BUILD_PYTHON "Build the MaterialX Python package from C++ bindings. Requires Python 2.7 or greater." OFF)
 option(MATERIALX_BUILD_VIEWER "Build the MaterialX Viewer." OFF)
 option(MATERIALX_BUILD_DOCS "Create HTML documentation using Doxygen. Requires that Doxygen be installed." OFF)
+
+option(MATERIALX_BUILD_GEN_GLSL "Build the GLSL shader generator back-end." ON)
+option(MATERIALX_BUILD_GEN_OSL "Build the OSL shader generator back-end." ON)
```

Comment:
@bernardkwok Are there situations in which you've found it valuable to disable shader generation in GLSL or OSL?  Because MaterialXGenGlsl and MaterialXGenOsl have no external library dependencies, I've been imagining that it's safe to build these modules in all configurations, but there may be use cases I'm not aware of.

Reply:
Hi @jstone-lucasfilm, This is part of make as many modules as possible options to accommodate web. The idea is both are never required in this situation and a new OpenGL ES generator will be added.

Reply:
Thanks for the clarification, @bernardkwok, and that makes sense.